### DISCUSSION TITLE
Can I keep selection when click outside the grid.

### DISCUSSION THREAD

**Discussion by chairedevil** [nick-keller/react-datasheet-grid#73] [Q&A] [answered]
![スクリーンショット 2021-10-27 155220](https://user-images.githubusercontent.com/22753035/139020226-b979240a-49df-4a45-ac03-966a476d7dc0.jpg)

I want to get the selected data when I click on the button[Get selected info] outside the grid.
but, when I click on the button the selection was eliminated.
```
        <button
          onClick={() => {
            console.log(sheetRef.current)
            console.log(sheetRef.current?.selection && sheetRef.current?.selection())
          }}
        >
          Get selected info
        </button>
```
so, sheetRef.current?.selection will be null.
I solve this problem by ref that update selection info every time onSelectionChange was fired
but in the future, I have to build a function that can select on the first grid.
And then, the related information selected in the first grid is displayed in the second grid.
Therefore, it is necessary to display which data in the first grid is selected.

Thank you so much

**Comment by nick-keller** [comment, marked as answer]
Clicking the button blurs the grid before you have time to read the selection.
You can keep track of the selection in a state and then read the state when you need it:
```ts
const [lastSelection, setLastSelection] = useState(null)

return <DataSheetGrid onSelectionChange={selection => setLastSelection(selection ?? lastSelection)} />
```

**Reply by chairedevil** [reply]
> const [lastSelection, setLastSelection] = useState(null)
> 
> return <DataSheetGrid onSelectionChange={selection => setLastSelection(selection ?? lastSelection)} />

I'm using the method above right now.

but in the future, I need to display which data is selected (color border). to let the user know that they are selecting
Do I need to use CSS in order to achieve what I want?
something like. 
```
dsg-row:nth-child(lastSelection.min.row) {
    .dsg-cell:nth-child(lastSelection.min.col) {
        border: xxxxx
    }
}
```

thank you

**Reply by nick-keller** [reply]
I'm not sure I understand your problem correctly. But you can use `cellClassName` on a column to give each cell a specific css class based on some condition by using a function.

**Reply by chairedevil** [reply]
![image](https://user-images.githubusercontent.com/22753035/139621577-4414de81-4f79-42b8-add3-816e558e30b8.png)
I want to show div(dsg-active-cell) even if it doesn't focus on the sheet.

**Reply by chairedevil** [reply]
@nick-keller

**Reply by nick-keller** [reply]
- You can apply a CSS class to the desired cells using `cellClassName`.
- You can manually select cells within a grid using a `ref`

Today you cannot have multiple grids focused at the same time, just like you cannot have multiple inputs focused.

If you need further help, or if you have a feature request please respond with a detailed explanation and I'll try to address it!

**Reply by kivircik-parantez** [reply]
I also need the same feature, I want to put an input on the grid and operate on the selections, but even though it keeps the selection in the background, even though I can make the selection again with `ref` as you said, the operation is not possible because the focus goes there all the time. I added a screenshot. Every time I click on an input, it takes the focus on grid. Is it possible to remove outside event listeners maybe?  @nick-keller 

![image](https://user-images.githubusercontent.com/54854223/211680582-83aa0fbb-b409-44c0-8428-8964c837a94b.png)
